scripts/lib/CIME/case/case.py

```diff
@@ -84,6 +84,9 @@ def __init__(self, case_root=None, read_only=True):
 
         if case_root is None:
             case_root = os.getcwd()
+
+        expect(not os.path.isdir(case_root) or os.path.isfile(os.path.join(case_root,"env_case.xml")), "Directory {} does not appear to be a valid case directory".format(case_root))
```

Comment:
At first blush I was a little confused by the logical statement... but this error will only be triggered if `case_root` exists but `{case_root}/env_case.xml` does not? And when `case_root` does not exist, there's already logic to handle whether that's allowed or not?

Reply:
If the case_root can be created with mkdir -p it will be by create_newcase and create_test.
In master I was able to use the record_cmd function to confirm that the directory exists if expected.   In maint-5.6 I was only able to put this check in xmlchange itself.